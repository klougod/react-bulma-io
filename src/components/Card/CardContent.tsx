import React from 'react'
import PropTypes from 'prop-types'

export const CardContent = ({ className, children }: any) => (
  <header className={`card-content ${className}`}>{children}</header>
)

CardContent.defaultProps = {
  className: ''
}

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default CardContent
