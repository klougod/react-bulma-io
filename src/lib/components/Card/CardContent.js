import React from 'react'
import PropTypes from 'prop-types'

export const CardContent = ({ className, children }) => (
  <header className={`card-content ${className}`}>
    { children }
  </header>
)

CardContent.defaultProps = {
  className: ''
}

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardContent