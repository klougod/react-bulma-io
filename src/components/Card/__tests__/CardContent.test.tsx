import React from 'react'
import Enzyme from 'enzyme'

import CardContent from '../CardContent'

describe('CardContent component test', function () {
  it('should render without crash without props', function () {
    const cardContentMock = Enzyme.shallow(<CardContent />)
    expect(cardContentMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const cardContentMockProps = {
      className: 'cardContent CLass'
    }
    const cardContentMock = Enzyme.shallow(
      <CardContent {...cardContentMockProps} />
    )
    expect(cardContentMock).toMatchSnapshot()
  })
})
