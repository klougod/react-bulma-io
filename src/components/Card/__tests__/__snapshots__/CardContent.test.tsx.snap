// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent component test should render without crash with props 1`] = `
<header
  className="card-content cardContent CLass"
/>
`;

exports[`CardContent component test should render without crash without props 1`] = `
<header
  className="card-content "
/>
`;
